app/javascript/components/sessions/SignIn.jsx

```diff
@@ -7,45 +7,15 @@ import Col from 'react-bootstrap/Col';
 import Row from 'react-bootstrap/Row';
 import { Link, useNavigate } from 'react-router-dom';
 import { useMutation, useQueryClient } from 'react-query';
+import {useCreateSession} from "../../hooks/mutations/sessions/useCreateSession";
 
 export default function SignIn() {
-  // POST request to server to create a session using email and password from Sign In form.
-  async function createSession(sessionUser) {
-    const response = await fetch('/api/v1/sessions.json', {
-      method: 'POST',
-      headers: {
-        'Content-Type': 'application/json',
-        'Accept': 'application/json',
-      },
-      body: JSON.stringify({
-        session: {
-          email: sessionUser.email,
-          password: sessionUser.password,
-        },
-      }),
-    });
-    if (!response.ok) throw new Error('Email or password is incorrect.');
-    return response.json();
-  }
-
-  // React Query useMutation
-  const navigate = useNavigate();
-  const queryClient = useQueryClient();
-  const { mutate, error } = useMutation(createSession, {
-    // Re-fetch the current_user and redirect to homepage if Mutation is successful.
-    onSuccess: () => {
-      queryClient.invalidateQueries('current_user');
-      navigate('/rooms');
-      console.log('mutate success');
-    },
-    onError: (error) => {
-      console.log('mutate error', error);
-    },
-  });
+  const { mutate } = useCreateSession();
 
   // Form handling needs access to mutate method from useMutation
   const { register, handleSubmit, formState: { errors } } = useForm();
   const handleSignIn = (sessionUser) => mutate(sessionUser);
```

Comment:
We can export this `handler function` from the `useCreateSession` mutation hook and make it a pattern where we call such function witha  prefix of `handle[Subject]`.

Reply:
Agreed.
We could replace line 13 and 17 with 
`const { handleSignIn } = useCreateSession();`

I thought it made sense to let it in the component since it deals directly with the form - but this looks cleaner.